Tallies/TallyClerks/MGxsClerk_class.f90

```diff
@@ -0,0 +1,887 @@
+    end do
+    call outFile % endArray()
+
+    name = 'transportOutScatter'
+    call outFile % startArray(name, resArrayShape)
+    do i=1,product(resArrayShape)
+      call outFile % addResult(transOS(1,i),transOS(2,i))
+    end do
+    call outFile % endArray()
+
+    name = 'nu'
+    call outFile % startArray(name, resArrayShape)
+    do i=1,product(resArrayShape)
+      call outFile % addResult(nu(1,i),nu(2,i))
+    end do
+    call outFile % endArray()
+
+    name = 'chi'
+    call outFile % startArray(name, resArrayShape)
+    do i=1,product(resArrayShape)
+      call outFile % addResult(chi(1,i),chi(2,i))
+    end do
+    call outFile % endArray()
+
+    ! Modify the shape of the result array to print matrices
+    resArrayShape(1) = resArrayShape(1) * self % energyN
+    name = 'P0'
+    call outFile % startArray(name, resArrayShape)
+    do i=1,product(resArrayShape)
+      call outFile % addResult(P0(1,i),P0(2,i))
+    end do
+    call outFile % endArray()
+
+    name = 'P1'
+    call outFile % startArray(name, resArrayShape)
+    do i=1,product(resArrayShape)
+      call outFile % addResult(P1(1,i),P1(2,i))
+    end do
+    call outFile % endArray()
+
+    name = 'prod'
+    call outFile % startArray(name, resArrayShape)
+    do i=1,product(resArrayShape)
+      call outFile % addResult(prod(1,i),prod(2,i))
+    end do
+    call outFile % endArray()
+
+    ! Deallocate from memory
+    deallocate(capt, fiss, transFL, transOS, nu, chi, P0, P1, prod)
+
+    ! If high order scattering is requested, print the other matrices
+    if (self % PN) then
+
+      call self % processPN(mem, P2, P3, P4, P5, P6, P7)
+
+      name = 'P2'
+      call outFile % startArray(name, resArrayShape)
+      do i = 1,product(resArrayShape)
+        call outFile % addResult(P2(1,i),P2(2,i))
+      end do
+      call outFile % endArray()
+
+      name = 'P3'
+      call outFile % startArray(name, resArrayShape)
+      do i = 1,product(resArrayShape)
+        call outFile % addResult(P3(1,i),P3(2,i))
+      end do
+      call outFile % endArray()
+
+      name = 'P4'
+      call outFile % startArray(name, resArrayShape)
+      do i = 1,product(resArrayShape)
+        call outFile % addResult(P4(1,i),P4(2,i))
+      end do
+      call outFile % endArray()
+
+      name = 'P5'
+      call outFile % startArray(name, resArrayShape)
+      do i = 1,product(resArrayShape)
+        call outFile % addResult(P5(1,i),P5(2,i))
+      end do
+      call outFile % endArray()
+
+      name = 'P6'
+      call outFile % startArray(name, resArrayShape)
+      do i = 1,product(resArrayShape)
+        call outFile % addResult(P6(1,i),P6(2,i))
+      end do
+      call outFile % endArray()
+
+      name = 'P7'
+      call outFile % startArray(name, resArrayShape)
+      do i = 1,product(resArrayShape)
+        call outFile % addResult(P7(1,i),P7(2,i))
+      end do
+      call outFile % endArray()
+
+      ! Deallocate from memory
+      deallocate(P2, P3, P4, P5, P6, P7)
```

Comment:
Does it do at the end of the calculation only? I added this line because MGxss can get very expensive, so I wanted to free some memory in between printing in the output file the xss and the scattering matrices. This is also because the output file itself takes a lot of memory (with the huge tape written all at once), and I had my simulation failing while writing the file in the past. I would keep this here for now, at least until we modify the output file writing.

Reply:
I will quickly add this comment, and then I think we can go ahead and merge!